### PR TITLE
Add makezero linter.

### PR DESCRIPTION
**Pull request by mbobrovskyi**
#### What type of PR is this?
/kind bug

#### What this PR does / why we need it:
To find slice declarations with non-zero initial length.

#### Which issue(s) this PR fixes:

Fixes #

#### Does this PR introduce a user-facing change?

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/jobframework/setup_test.go [hunk 1]
```diff
@@ -120,8 +120,8 @@ func TestSetupControllers(t *testing.T) {
 				},
 				MapperProvider: func(*rest.Config, *http.Client) (apimeta.RESTMapper, error) {
 					gvs := make([]schema.GroupVersion, len(tc.mapperGVKs))
-					for _, gvk := range tc.mapperGVKs {
-						gvs = append(gvs, gvk.GroupVersion())
+					for i, gvk := range tc.mapperGVKs {
+						gvs[i] = gvk.GroupVersion()
 					}
```
Comment (trasc) [inline pkg/controller/jobframework/setup_test.go:125]:
```suggestion
gvs := slices.Map(tc.mapperGVKs, func(gvk *schema.GroupVersionKind ) schema.GroupVersion {return gvk.GroupVersion()} )
```
from `kueue/pkg/util/slices`

Comment (mbobrovskyi) [reply pkg/controller/jobframework/setup_test.go]: Done

**Review by trasc — COMMENTED** [review]
/lgtm

**Pull request merged**